Nest 2nd version - 02 Introduction (#4)
raven-os/getting-started: nest/02.introduction.md

It is similar to an app store for a smartphone, or the Windows Store for example.
One of the key differences between those and Nest (and most package managers), is that the upgrading must be done manually.
On a smartphone, the applications are usually automatically upgraded.
With Nest, it's up to the user to upgrade their installed packages whenever they want.

Currently, to use Nest, one must use its Command Line Interface (CLI).
This might sound intimidating for someone who has never seen a terminal or typed any command before, but if that's the case, don't worry! This guide is here to help you feel at home.

## 1.2 Packages and dependencies
It is common for a package to require other ones to work properly.
For example, let's take a software that allows its users to send messages to each other, over the Internet.
That software will need a way to send messages over the Internet.
To do so, it will either implement its own solution, or use an existing library for that purpose.
In the latter case, the software **requires** that library, because without it, the messages simply can't be sent.
In other words, that library is a **dependency** of the software.

> Whenever you install a package via a package manager (and Nest as well), it will check if the package in question has any dependencies.

Reviewer: ```suggestion
Whenever you install a package via Nest, it will (like most package managers) check if the package in question has any dependencies.
```
Reversing it gives a clearer message: Nest does this. The fact that most other package managers do it to is secondary.

If it does, Nest will install the dependencies that aren't already installed.

> So, when you install a package, don't be surprised if you see other packages getting installed too, they are just dependencies needed for that package to work properly.

Reviewer: Does this sentence have any purpose ? I don't think it conveys any information.
Its tone is also a bit weird.

Author: Its goal is to explain that when you install a package, you can see other packages getting installed too, which are its dependencies
If I'm totally new to Nest and package managers in general, I'd be surprised if when I'm installing a single package, other packages are installed with it. Even if after reading this section you should know that they are dependencies, I figured it's not that big of a deal to explicitly write it

Reviewer: Hum, I'd change the sentence above then.

Reviewer: How about this?
```suggestion
In other words, when asking Nest to install a package, Nest will install other packages alongside it to satisfy its dependencies.
```
This way it's obvious you're restating the same thing than the previous statement and we keep a neutral tone.

## 1.3 Repositories
Packages are stored in repositories.

> Within each repository, there are a certain number of categories, and a package has to belong to one, and only one of them.

Reviewer: ```suggestion
Each repository has one or more categories, and each package belongs to a specific one of these.
```
(This is phrased as a requirement, not a fact, it may be a good idea to change that too.)

Author: you wrote `each package has to belongs`, but I'm guessing you meant `each package belongs`?

Reviewer: Woops!
I've edited the comment above.
I wanted to keep it as a requirement since you wrote it like that originally, but I'm fine with this version too.

> The categories are a way to distinguish packages with the same name, but having different purposes.

Reviewer: ```suggestion
Categories are used to distinguish between packages that have the same name, but have different purposes.
```

> This means that we can have two packages with the same name, in the same repository but not in the same category, coexisting together, without any issue.

Reviewer: ```suggestion
This means two packages with the same name and in the same repository but not in the same category can coexist without any issue.
```
Again, this seems to be stated as a requirement.

Author: your sentence gets a bit long without any comma, how does this sound?
```suggestion
This means that two packages with the same name and in the same repository, but not in the same category, can coexist without any issue.
```

Reviewer: I'd argue the opposite: you use quite a lot of commas.
I don't think they help with the legibility of the sentence. I find the one before "can coexist" weird, and removing it makes the previous one quite weird too. I'd ask a native English speaker but I don't have one around, so I'll let you decide what you prefer.

`stable` is the repository enabled by default for Nest. It contains stable packages that are safe to use, and is maintained by the Raven-OS team.

## 1.4 Package name syntax

> In Nest, the package names are structured in the following way: `repository::category/package#version`.

Reviewer: ```suggestion
In Nest, the package names are structured in the following way:
```repository::category/package#version```
```
I think having the syntax of the package name helps glimpse the important information of this part faster.

If no version is specified, the latest one available is taken by default.

> However, most parts of this syntax may be omitted if it matches **one** and only **one** package.

Reviewer: ```suggestion
However, parts of this syntax may be omitted if there's only one package that matches the remaining criteria.
```

> In a way, the only real mandatory part of this syntax is the package name itself. That means you can use:

Reviewer: ```suggestion
The only real mandatory part of this syntax is the package name itself. The other parts are only necessary to resolve ambiguities. That means you can use:
```
(I think this answers the obvious question: "in what way?")

* `repository::package`, if that package is the only one with its name in the whole repository.
* `category/package`, if the package's name is unique in this specific category, and if that category belongs to a single repository used by Nest.
* `package`, if that package is the only one with its name amongst all the repositories used by Nest.

> If you're not sure on the repository, category, or name of a package, you can check on [Raven-OS official website](https://stable.raven-os.org).

Reviewer: ```suggestion
If you're not sure on the repository, category, or name of a package, you can look it up on the [Raven-OS official website](https://stable.raven-os.org).
```

It lists all the packages in the repository `stable` of Nest, as well as the category they belong to, the latest version released, and a short description of what they provide.

> The repository `stable` is one of the official repositories available for Nest, and is enabled by default.

Reviewer: This information was already given in the previous section. However not the exact same additional info was given.
Also  this means that the previous sentence ("If you're not sure on the repository, category, or name of a package...") is not entirely accurate, since the solution only works for packages within the `stable` repository.

Author: nice catch! I will remove the last sentence and remove `the repository` in the first one

# 1 Introduction
## 1.1 What is a package manager?
A package is a compressed file archive containing all the files required to install, uninstall and update a particular software or application.

A package manager allows the user to install, upgrade or remove software packages on their OS, in other words, to **manage packages**.
It is similar to an app store for a smartphone, or the Windows Store for example.
With Nest, it's up to the user to upgrade their installed packages whenever they want.

Currently, to use Nest, one must use its Command Line Interface (CLI).
This might sound intimidating for someone who has never seen a terminal or typed any command before, but if that's the case, don't worry! This guide is here to help you feel at home.

## 1.2 Packages and dependencies
It is common for a package to require other ones to work properly.
For example, let's take a software that allows its users to send messages to each other, over the Internet.

> That software will need a way to send messages over the Internet, so either it has its own way to do it, either it will use an existing library for that purpose.

Reviewer: ```suggestion
That software will need a way to send messages over the Internet. To do so it will either implement it's own solution, or use an existing library for that purpose.
```

> In the latter case, the software **requires** that library, because without it, the messages simply can't be sent.

Reviewer: ```suggestion
In the later case, the software **requires** that library, because without it, the messages simply can't be sent.
```

Author: `latter` is correct actually
https://www.dictionary.com/browse/latter
https://www.dictionary.com/browse/later

Reviewer: Indeed.

In other words, that library is a **dependency** of the software.

Whenever you install a package via a package manager (and Nest as well), it will check if the package in question has any dependencies.

> And if it does, it will install only the dependencies that are not already installed.

Reviewer: ```suggestion
If it does, Nest will install the dependencies that aren't already installed.
```
(the double it could be confusing)

# 1 Introduction
## 1.1 What is a package manager?

> A package is a compressed file archive containing all of the files coming with a particular software or application.

Reviewer: ```suggestion
A package is a compressed file archive containing all the files required by a particular software or application.
```

Reviewer 2: ```suggestion
A package is a compressed file archive containing all the files required to install, uninstall and update a particular software or application.
```

> A package manager allows the user to install, upgrade or remove software packages on their OS, in other words, to **manage the packages**.

Reviewer: ```suggestion
A package manager allows the user to install, upgrade or remove software packages on their OS, in other words, to **manage packages**.
```

> It is similar to an app store for any smartphone, or the Windows Store for example. One of the main differences between those and Nest (and most package managers), is that the upgrading must be manually commanded.

Reviewer: ```suggestion
It is similar to an app store for a smartphone, or the Windows Store for example.
One of the main differences between those and Nest (and most package managers), is that the upgrading must be done manually.
```
or
```suggestion
It is similar to an app store for a smartphone, or the Windows Store for example.
One of the key differences between those and Nest (and most package managers), is that the upgrading must be done manually.
```

The second sentence sounds wonky to me.

> On a smartphone, the applications are usually automatically upgraded overnight when you're charging your phone, if an update is available.

Reviewer: I think this actually true.

Author: Did you meant to say `isn't true`, or were you just agreeing with what I wrote?

Reviewer: yeah sorry, I meant untrue.

Author: maybe I shouldn't have based myself on my experience since I've only had Android smartphones. I'll remove that sentence then

Reviewer: Idk, I've got an android too and it seems to want to update whenever I have a stable and free internet connection. I think something like this would be ok:
```suggestion
On a smartphone, the applications are usually automatically upgraded.
```

> Here, it's up to the user to upgrade their installed packages whenever they want or need to.

Reviewer: ```suggestion
With Nest, it's up to the user to upgrade their installed packages whenever they want.
```
I think you are never 'forced" to update a package.

> To use Nest, the only interface available for now is the Command Line Interface (CLI).

Reviewer: ```suggestion
Currently, to use Nest, one must use its Command Line Interface (CLI).
```

> It might be a bit nerve racking for someone who has never seen a terminal or typed any command before, but don't worry, this guide is here to help you feel comfortable!

Reviewer: ```suggestion
This might sound intimidating for someone who has never seen a terminal or typed any command before, but if that's the case, don't worry! This guide is here to help you feel at home.
```

## 1.2 Packages and dependencies

> Often, a package requires another one to work properly.

Reviewer 2: ```suggestion
It is common for a package to require other ones to work properly.
```

> Say, for example, someone develops a messenging application, which relies on encrypted messages to provide communication between two users.

Reviewer 2: Instead of saying `someone develops` (which could lead to someone asking himself "I'm not a developer, do I need dependencies?"), maybe use `someone is using a messaging application` or anything that applies to anyone.

> This application will probably use a package, written by someone else, to encrypt the messages sent with their app.

Reviewer: This example is really complicated. Maybe something simpler a package using a programming language or a GUI toolkit would be clearer?

Reviewer 2: I think using the discord app as an example could be fine, it has some flashy dependencies that seem obvious even for a non-technical user: a library to render web pages, another to send requests over the internet etc. 

(I don't think you need to be specific, being vague is enough)

Reviewer: Yes, I'd just avoid the use of concepts that sound complicated (encryption).

We say that such messenging application **depends** on that encryption package, because without that package, the application would probably not work at all.
Now, if you want that application to work properly, you'll also need to install the package they used to encrypt messages.

If a package is required by another one to work properly, that package is called a **dependency**.
Whenever you install a package via a package manager (and Nest as well), it will check if the package in question has any dependencies.

> And if it does, it will install only the dependencies that are not already installed on the OS.

Reviewer 2: ```suggestion
And if it does, it will install only the dependencies that are not already installed.
```

So, when you install a package, don't be surprised if you see other packages getting installed too, they are just dependencies needed for that package to work properly.

## 1.3 Repositories
Packages are stored in repositories.

> Think of a repository as a big folder containing the packages.

Reviewer: "folder" is not a unix term. Also I'm not sure "repository" needs analogy given the original meaning of the word.

Within each repository, there are a certain number of categories, and a package has to belong to one, and only one of them.
The categories are a way to distinguish packages with the same name, but having different purposes.
This means that we can have two packages with the same name, in the same repository but not in the same category, coexisting together, without any issue.

[//]: # (TODO: Add link to the section 3.1)
For now, all you need to know is that there are official and unofficial repositories for Nest.
Only `stable`, which is an official repository, is enabled by default.
This repository contains all the packages that are considered stable and safe to use.

> The repositories will be more detailed in [a future section]().

Reviewer: You should name the section specifically.

## 1.4 Package name syntax
[//]: # (TODO: add link to the section 3.3)

> The packages names are represented in the following way by Nest: `repository::category/package#[version|version_req]` (*version_req* means *version requirement*).

Reviewer: ```suggestion
In Nest, the package names are structured in the following way: `repository::category/package#[version|version_req]` (*version_req* means *version requirement*).
```

Reviewer 2: I consider version requirements to be an advance concept that shouldn't be explained nor mentioned here. Sticking to simply `version` is probably enough as of now, and version_req be explained in further details later?

What do you guys think?

Reviewer: Might be a good idea. May have the pitfall that a person searching for "version requirement" will think it's impossible so perhaps a footnote or something may be a good idea.

Author: I don't see too much difference between having a footnote and what I had written.
I agree with @Arignir , it's better to remove it from this section

> The concept of version requirement will be detailed in [this section]().

Reviewer: Name the section.

> If you don't specify any version or version requirement, the latest one available will be installed by default.

Reviewer: ```suggestion
If no version or version requirement is specified, the latest one available is taken by default.
```
(I think this syntax is used for other things than installing)

Reviewer 2: (It is)

However, most parts of this syntax may be omitted if it matches **one** and only **one** package.
In a way, the only real mandatory part of this syntax is the package name itself. That means you can use:
* `repository::package`, if that package is the only one with its name in the whole repository.
* `category/package`, if the package's name is unique in this specific category, and if that category belongs to a single repository used by Nest.
* `package`, if that package is the only one with its name amongst all the repositories used by Nest.

> If you're not sure on the repository, category, or name of a package, you can check on [Raven-OS official website](https://packages.raven-os.org).

Reviewer 2: That URL is wrong now, as https://packges.raven-os.org is now down. You can use https://stable.raven-os.org instead, and modify the following sentences accordingly.

It lists all the packages in the official repositories for Nest.

> Meaning you can see the repository and category they belong to, the latest version released for Nest, and a short description of what they are for.

Reviewer 2: `what they are for`? Is that English? What does it mean? :o

Reviewer: "à quoi ils servent" in french. Maybe replace that with "what they provide"?
I don't like how this sentence starts with a "meaning". 
I'd replace this and the previous sentence with something like this:
"It lists all the packages in the official repositories for Nest as well as the repository and category they belong to, the latest version released, and a short description of what they provide."

## Conclusion
[//]: # (TODO: add link to chapter 2)
And that's it for the introduction!

> Those were some basic concepts, and we strongly recommend you to understand them before moving on to the [next chapter](), where we'll start detailing how to use `nest`.

Reviewer: ```suggestion
Those were some fundamental concepts, and we strongly recommend you to make sure you understand them before moving on to the [next chapter](), where we'll start detailing how to use `nest`.
```
(might demotivate the reader with "basic")
